lib/handler/range.c

```diff
@@ -0,0 +1,119 @@
+#include <assert.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "h2o.h"
+
+#if __GNUC__ >= 3
+# define likely(x) __builtin_expect(!!(x), 1)
+# define unlikely(x) __builtin_expect(!!(x), 0)
+#else
+# define likely(x) (x)
+# define unlikely(x) (x)
+#endif
+
+#define CHECK_EOF()                             \
+    if (buf == buf_end) {                       \
+        *ret = -2;                              \
+        return NULL;                            \
+    }
+
+#define EXPECT_CHAR(ch)                         \
+    CHECK_EOF();                                \
+    if (*buf++ != ch) {                         \
+        *ret = -1;                              \
+        return NULL;                            \
+    }
+
+size_t *process_range(h2o_mem_pool_t *pool, h2o_iovec_t *range_value, size_t file_size, size_t* ret)
+{
+    size_t range_start = -1, range_count = 0;
+    char *buf = range_value->base, *buf_end = buf + range_value->len;
+    int good_range=1;
+    H2O_VECTOR(size_t) ranges = {};
+    
+    EXPECT_CHAR('b');
+    EXPECT_CHAR('y');
+    EXPECT_CHAR('t');
+    EXPECT_CHAR('e');
+    EXPECT_CHAR('s');
+    EXPECT_CHAR('=');
```

Comment:
simply doing `if (range_value->len < 6 || memcmp(range_value->base, "bytes=", 6) != 0)` might be sufficient